src/App.tsx

```diff
@@ -1,24 +1,239 @@
+import { TempTodo } from './components/TempTodo';
+import { Status } from './types/Status';
+import { ErrorMessages } from './utils/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 11457;
+const initialTodos: Todo[] = [];
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>(initialTodos);
+  const [filterBy, setFilterBy] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NoError);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [isToggled, setIsToggled] = useState(false);
+
+  const filteredTodos = getFilteredTodos(todoList, filterBy);
+
+  useEffect(() => {
+    postService.getTodos(USER_ID)
+      .then(setTodoList)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LoadError);
+      });
+    const timerId = setInterval(() => {
+      setErrorMessage(ErrorMessages.NoError);
+    }, 3000);
+
+    return () => {
+      clearInterval(timerId);
+    };
+  }, []);
+
+  const allTodosId = todoList.map(({ id }) => id);
+
+  const completedTodosId = todoList
+    .reduce((total: number[], current) => (current.completed
+      ? [...total, current.id]
+      : total), []);
+
+  const hasCompletedTodosCount = !!completedTodosId.length;
+
+  const activeTodosCount = todoList
+    .filter(({ completed }) => completed === false).length;
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingId(prevTodoIds => [...prevTodoIds, todoId]);
+    setIsLoaderActive(true);
+
+    return postService.deleteTodo(todoId)
+      .then(() => {
+        setTodoList(currentList => currentList
+          .filter(todo => todo.id !== todoId));
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.DeleteError);
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    completedTodosId
+      .forEach((id) => {
+        handleDeleteTodo(id);
+      });
+  };
+
+  const addNewTodo = (todo: Todo) => {
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    });
+
+    setErrorMessage(ErrorMessages.NoError);
+    setLoadingId([0]);
+    setIsLoaderActive(true);
+
+    postService.createTodo(todo)
+      .then(newTodo => {
+        setTodoList(currentTodos => [...currentTodos, newTodo]);
+        setTitle('');
+        setTimeout(() => {
+          setIsLoaderActive(false);
+          setLoadingId([]);
+        }, 3000);
+      })
+      .catch(() => {
+        setIsLoaderActive(false);
+        setLoadingId([0]);
+        setErrorMessage(ErrorMessages.AddError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
```

Comment:
```suggestion
    postService.createTodo(todo)
      .then(newTodo => {
        setTodoList(currentTodos => [...currentTodos, newTodo]);
        setTitle('');
      })
      .catch(() => {
        setErrorMessage(ErrorMessages.AddError);
      })
      .finally(() => {
        setTempTodo(null);
        setLoadingId([]);
        setIsLoaderActive(false);
      });
```
I have no idea why in a successful case you remove the loader only after 3 seconds 🤷‍♂️ 
It looks strange, and why do you do `setLoadingId([0])` in case of error, as it will be an infinite spinner for tempTodo,  but in finally block you remove it.